bookmarket-backend/src/main/java/edu/oakland/sophomoreproject/dependencies/sqlite/comments/CommentsTableAccessor.java

```diff
@@ -7,29 +7,82 @@
 import org.springframework.stereotype.Component;
 import org.sqlite.SQLiteConfig;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Instant;
 import java.util.ArrayList;
 import java.util.List;
 
 @Component
 public class CommentsTableAccessor extends TableAccessor {
-	@Autowired
-	public CommentsTableAccessor(SQLiteConfig sqliteConfig) {
-		super(sqliteConfig);
-	}
-
-	public List<Comment> getAllCommentsForListing(int commentId) {
-		List<Comment> comments = new ArrayList<>();
-
-		// TODO
-
-		return comments;
-	}
-
-	/// this object is CommentWithoutId because we will get the ID from SQL after creating it
-	/// this can be done by doing `INSERT INTO comments ... RETURNING comment_id`
-	/// and parsing the `comment_id` column it returns with `results.getString("comment_id")`
-	/// also you don't need a `createReply` function since the only difference is whether `parentCommentId` is null or not
-	public void createComment(CommentWithoutId commentWithoutId) {
-		// TODO
-	}
+    @Autowired
+    public CommentsTableAccessor(SQLiteConfig sqliteConfig) {
+        super(sqliteConfig);
+    }
+
+    public List<Comment> getAllCommentsForListing(int listingId) throws SQLException {
+        List<Comment> comments = new ArrayList<>();
+        String sql = "SELECT * FROM comments WHERE listing_id = ? ORDER BY created_at";
+
+        try (Connection connection = getDatabaseConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setInt(1, listingId);
+            ResultSet results = statement.executeQuery();
+
+            while (results.next()) {
+                int commentId = results.getInt("comment_id");
+                String content = results.getString("content");
+                Instant createdAt = Instant.parse(results.getString("created_at"));
+                int creatorId = results.getInt("creator_id");
+                int parentListingId = results.getInt("listing_id");
+                Integer parentCommentId = results.getInt("parent_comment_id");
+                if (results.wasNull()) {
```

Comment:
I remember we talked about this in person. I can look into this after we merge the changes but I don't think this is exactly what we want.